app/services/google/scrape_service.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Google
+  class ScrapeService
+    attr_reader :keyword
+
+    def initialize(keyword_id:)
+      @keyword_id = keyword_id
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    # rubocop:disable Metrics/AbcSize
+    def call
+      keyword = Keyword.find(@keyword_id)
+      html = Google::SearchService.search!(keyword.content)
+      result = Google::ParseService.new(html).call
+
+      save_scrape_result keyword, result
+    rescue ActiveRecord::RecordNotFound => e
+      Rails.logger.error "[#{self.class.name}]: keyword doesn't exist
+    - keyword_id: #{@keyword_id}"
+
+      raise e
+    rescue GoogleScraperRuby::Errors::SearchServiceError => e
+      Rails.logger.error "[#{self.class.name}]: error while processing request
+    - keyword: #{keyword.content}
+    - error: #{e}"
+
+      raise e
```

Comment:
Shall we update the `failed` status for the Keyword here?

Reply:
I intended to handle to status update part in the [background job integration ticket](https://github.com/Goose97/google-scraper-ruby/issues/15)

Reply:
Shall we add the TODO so the reviewer will aware about your plan @Goose97

Reply:
yeah, sure. Fixed in e64878013a57db1b0ec6884073940edd47c5a74e